Added Getting Started guide for TFLite (#8136)
google/iree: docs/website/docs/index.md

needs of the datacenter and down to satisfy the constraints and special
considerations of mobile and edge deployments.

## User Guide

Use of IREE requires three main stages outlined in our 
[User Getting Started Guide](./getting-started/index.md). Overall these stages

> can be broken into import, compilation, and execution of a model using IREE.

Author: I mainly wanted to include this at head to divert non-developers to the Getting Started guide as quickly as possible. What is your thought on moving the workflow to the top.

> can be broken into import, compilation, and execution of a model using IREE.

Reviewer: Good point. Maybe worth splitting that sort of change into a separate PR.

Some ideas:
* We could move the "Project architecture" section down
* I think "Key features" and "Support matrix" are important to keep near the top so people can first learn what the project is and if it sounds useful, _then_ see how to use it
* The "Support matrix" section is a little _too_ sparse now... it could be condensed

> can be broken into import, compilation, and execution of a model using IREE.

Author: Overall having a toploaded section that is literally "Hey, if you just want to use things go here" with explanations remaining in the other section.

## Workflow overview

Specific examples outlining IREE's workflow can be found in the

> [User Getting Started Guide](./getting-started/index.md). Using IREE involves three

Reviewer 2: ```suggestion
[User Getting Started Guide](./getting-started/index.md). Using IREE involves the following
```

> general steps:

Reviewer 2: ```suggestion
the following steps:
```

1. **Import your model**

>     Work in your [framework of choice](./getting-started/index.md), then run your model

Reviewer 2: ```suggestion
    Developer your program using one of the [supported frameworks](./getting-started/#supported-frameworks), then run your model
```

>     through one of IREE's import tools.

Reviewer 2: ```suggestion
    using one of IREE's import tools.
```

> 2. **Select your [deployment configuration](./deployment-configurations)**

Reviewer 2: ```suggestion
2. **Select your [deployment configuration](./deployment-configurations).**
```

## Workflow overview

Specific examples outlining IREE's workflow can be found in the
[User Getting Started Guide](./getting-started/index.md). Using IREE involves four
general steps:

> 1. **Import your model**

Reviewer: there's a link to ./ml-frameworks a few lines down from here - can you redirect that and any others?

Author: Should be fixed up.

## Workflow overview

Specific examples outlining IREE's workflow can be found in the
[User Getting Started Guide](./getting-started/index.md). Using IREE involves three

> general steps:

Reviewer: > "three general steps"

> list with 4 items

🤔
